### PR TITLE
created missing predadj dependency for 'how adj' questions

### PR DESCRIPTION
**Pull request by anitzkin**
I found out that quesitons like "How crazy are you?" were not getting any _predadj dependency so I added one.  Except now, since they were getting a _subj() dependency, the SV-rule is firing twice, once for the _subj() and once for the _predadj() so I should correct that, although the redundancy isn't generating any incorrect relations, just redundancy.

### CODE REVIEW

**Comment by anitzkin** [comment]
This would be a good stage at which to merge my revsions so far.  That's why I made the commit yesterday, which covers all of my work since Jan 5 . . .

File: data/relex-semantic.algs [hunk 1]
```diff
@@ -331,6 +332,12 @@ DIR_OBJ1_1
 <F_L obj> = %
 <F_L obj> = <F_R linkR0 F_R ref>
 
+#TemplateActionAlg
+DIROBJ_Osm
+<LAB> = \O\.*\.m
```
Comment (linas) [inline data/relex-semantic.algs]: When adding new rules, its great to add one or two example sentences that trigger these rules

File: data/relex-semantic.algs [hunk 2]
```diff
@@ -534,7 +570,7 @@ PREDADJ_PREDET
 <F_L predet> != %
 <F_R subj links _predadj> != %
 =
-<F_R subj links _predadj> = <F_L predet>
```
Comment (linas) [inline data/relex-semantic.algs]: I cannot telll from here if you removed all uses of predet or not .. in any case, please make sure the wiki page that lists all of these relations is accurate and up-to-date.  (In particular, I think the back-n-forth confusion about quantity messed up the wiki page)

File: data/relex-semantic.algs [hunk 3]
```diff
@@ -772,6 +810,28 @@ HEAD_TO_CLAUSE_AND_WALL
...
+<F_R head-word> = %
+=
+<F_R head-word> = <F_R ref>
+
+#TemplateActionAlg
+DETERMINER_HEAD_WORD
+<LAB> = \D\.*
+<F_L head-word> = %
+=
+<F_L head-word> = <F_R head-word>
```
Comment (linas) [inline data/relex-semantic.algs]: ???  This seems deeply mysterious to me ... if you see a determiner, e.g. "the", and the noun on its right has a head-word, then you're going to set the determiners head-word to that?? What for? why would you want to do that?

File: data/relex-semantic.algs [hunk 4]
```diff
@@ -1339,19 +1407,27 @@ TOBE_TO_LINKS
...
+<LAB> = \Ju
+<F_R POS> = conjunction
+=
+<F_L ref links conj_prep> += <F_R ref>
+
+#TemplateActionAlg
+SO_THAT_STRUCTURE
+<LAB> = \EExk
+=
+<F_L str> = so_that
```
Comment (linas) [inline data/relex-semantic.algs]: This looks weird/wrong/bad to me: you are actually modifying the words in the sentence with this rule !??  Surely you don't want to modify the sentence itself!!

**Comment by linas** [comment]
OK, I'm going to merge this ... but I am really not happy about it. It is much too large a set of changes to review in one gulp.  It is essentially impossible for me to review what's going on here: its just one giant black box of stuff.   This is why I kept saying: merge early, merge often: it makes the changes small and reviewable for design and correctness.

I saw a number of serious issues that must be fixed. I have to now trust you to fix them.  Do NOT do this ever again!  This is really bad form.

So: almost none of the added rules have any sort of example sentences illustrating the conditions under which they trigger.  This will make it hard for future maintainers to figure out what is going on.  

My eyes glazed over before I could get to the bottom of this, but:

Some of the changes look weird -- adding head words to determiners? Why?  Modifying the words in the sentence?  Surely that cannot be right ... Based on this, I  suspect there are other issues in here, which is again .. a reason to make changes in much smaller, more manageable chunks.

**Comment by linas** [comment]
And .. are you even trying to run the unit tests?  The number of test failures, which was already unacceptably high, doubles or tripled!!   Aaron ... !! I'm flabbergasted.  Should I be reverting this in its entirety?  There is something seriously wrong in here ... 

The number of failing comparatives triples from a 75% pass rate to a 75% fail rate.
The number of failing equatives triples from a 75% pass rate to a 75% fail rate.
Extrapositions, ditto ... 

There now seem to be 100 fewer comparative test sentences?  ???

Before merging, I get this:

```
     [java] Test failed
     [java]    6 sentences failed
     [java]    40 sentences passed
     [java] Determiners: Tested 6 sentences, test passed OK
     [java] Time: Tested 8 sentences, test passed OK
     [java] Comparatives: Test failed
     [java]    22 sentences failed
     [java]    72 sentences passed
     [java] Equatives: Test failed
     [java]    30 sentences failed
     [java]    76 sentences passed
     [java] Extrapostion: Test failed
     [java]    35 sentences failed
     [java]    84 sentences passed
     [java] Conjunction: Test failed
     [java]    36 sentences failed
     [java]    97 sentences passed
     [java] Inquisitives: Test failed
     [java]    50 sentences failed
     [java]    180 sentences passed
     [java] Test failed
     [java]    50 sentences failed
     [java]    180 sentences passed
```

After merging, I get this:

```
     [java] Test failed
     [java]    16 sentences failed
     [java]    30 sentences passed
     [java] Determiners: Tested 6 sentences, test passed OK
     [java] Time: Tested 8 sentences, test passed OK
     [java] Comparatives: Test failed
     [java]    72 sentences failed
     [java]    22 sentences passed
     [java] Equatives: Test failed
     [java]    84 sentences failed
     [java]    22 sentences passed
     [java] Extrapostion: Test failed
     [java]    88 sentences failed
     [java]    31 sentences passed
     [java] Conjunction: Tested 45 sentences, test passed OK
     [java] Interrogatives: Test failed
     [java]    92 sentences failed
     [java]    126 sentences passed
     [java] Test failed
     [java]    92 sentences failed
     [java]    126 sentences passed
```

**Comment by linas** [comment]
Correction: the number of test failures did increase, but not as badly as in the last comment. Turns out the unit test was incorrectly counting pass-fail.  Before this commit, there was a total of 230 test sentences, with 50 fails and 180 passes.  After this commit, there are fewer tests: out of 218 test sentences, there are now 92 failures and 126 passes.

Why is the total number of test sentences going down?

**Comment by linas** [comment]
Here's the actual, true summary:
```
     [java] Stanford compatibility: Test failed; out of 47 sentences total,
     [java]    17 sentences failed
     [java]    30 sentences passed
     [java] Determiners: Tested 6 sentences, test passed OK
     [java] Time: Tested 2 sentences, test passed OK
     [java] Comparatives: Test failed; out of 86 sentences tested,
     [java]    72 sentences failed
     [java]    14 sentences passed
     [java] Equatives: Test failed; out of 12 sentences tested,
     [java]    12 sentences failed
     [java]    0 sentences passed
     [java] Extrapostion: Test failed; out of 13 sentences tested,
     [java]    4 sentences failed
     [java]    9 sentences passed
     [java] Conjunction: Tested 14 sentences, test passed OK
     [java] Interrogatives: Test failed; out of 85 sentences tested,
     [java]    4 sentences failed
     [java]    81 sentences passed
     [java] Test failed; out of a total of 218 test sentences,
     [java]    92 sentences failed
     [java]    126 sentences passed
```

**Comment by anitzkin** [comment]
Okay, one issue at a time . . .

First of all, the unit test: when I run it, virtually all the sentences
that fail are comparatives (and 'equatives' which are part of the same
collection, because I haven't gotten to them yet.  They will all fail until
I go through each one, because there's not a comparative sentence
that isnot affected by some of my changes.  I have more-or-less finished
the revisions that are not connected to comparison; that's why
I thought it would be good to merge now.

As for why you would want to give a determiner the head-word of the noun, I
can't remember for sure why I needed to do that at that time, but I suspect
it was so I could have a quantification relation apply to the noun it
quantifies rather than any of the intervening stuff.  Or there was some
other case involving a word that is linked directly to the determiner
rather than the noun that it truly modifies.  This kind of thing is pretty
common in LG.  There are a lot of linkages incorrectly based on proximity
rather than logical relationships.

As for, why I would want to 'change the sentence' by constructing polywords
-- how can you even ask? You already have polywords
installed in relex, fairly randomly.  Why do you believe how_much should be
treated as one word but not so_that?  I've only made such
constructions in the rare cases when it seemed necesasry to get the
dependecies right.  In fact, I would be much happer if it were easier
to do this kind of thing.  I keep having to write algorthms to work-around
stuff that would bemore easily handled by changing the surface
structure of the sentence to better reflect its implicit structure.  There
is lots and lots of elision in the unit test sentences, which is almost
impossible to handle well with relex.  lots of implicit constitutents.
Also, a few of the unit test sentences are unintentionally un-gramatical or
semantically incoherent.  Although, strangely, one or two are commented as
being un-grammatical when they are not . . . Anyway, filling in some of the
gaps or making a few more poly-words helps get the logic of the dependency
relations right.

Cleaning up the relex rule file, especially the comments, and the wiki are
on my to do list.  I just thought it made more sense to finish working
through the comparatives first.  Keep in  mind that the relex rule file was
already an egregious disaster when I found it. More so. There were more
rules without comments and examples then, than there are now, because I've
commented copiously, and with more explanations than anyone else.  Also,
I've tried to ameliorate the fact that tons of rules were mis-named or in
strange sections.  You had people referring to words like 'that' and
'where' as prepositions, and adverbs being charactarized as direct-objects,
etc. and more than half of the rules had no example sentences or
explanations.  The grammar was so bad that even direct and indirect-objects
were confused with each other.  And that's just for starters.  I'm trying
to pull the poor thing together, but it's going to take more work.

**Comment by anitzkin** [comment]
Also, Linas, I should add, that the only reason the unit test "passed"
before, is because the people who updated it were simply putting in
whatever relex spit out as the unit test conditions, no matter how
obviously wrong.  it was my pleasure to discover for example that our most
recent disappeariing intern, Matthew, had done just that, and then simply
declined to correct any of the errors we pointed out to him, and didn't
tell anyone that he was at a loss, and as far as I can tell, he wasn't the
first one to do this.  It appears as if I have devastated the unit test,
because I'm not just cutting and pasting bad relex output into it.

**Comment by linas** [comment]
So I was in a tizzy ...


> As for why you would want to give a determiner the head-word of the noun,
> I
> can't remember for sure why I needed to do that at that time,


which is why placing example sentences into the algs file is a good iidea:
Which is exactly why I was tearing my hair out.


> As for, why I would want to 'change the sentence' by constructing
> polywords
> -- how can you even ask? You already have polywords
> installed in relex,


But not in the algs file. Yes, I suppose they could be put there.  Its just
surprising.  And unexplained

Look: the whole algs file is wildly under-documented. If the problem is
that LG and relex are doing random things that are poorly thought out and
half-implemented, then answer to this problem is not going to be to add yet
more random, crazy, undocumented functions.

stuff that would bemore easily handled by changing the surface
> structure of the sentence to better reflect its implicit structure.


Is this a comment about LG? Or something else?


> There
> is lots and lots of elision in the unit test sentences, which is almost
> impossible to handle well with relex. lots of implicit constitutents.
>

'elision' == elided words? e.g. "zero copula" or "zero that"? That's an
issue that I've been wracking my brains on.  I have not found any solution,
although I once thought I'd glimpsed it ...


> Also, a few of the unit test sentences are unintentionally un-gramatical
> or
> semantically incoherent.


None of them were written by native English speaker.


> Keep in mind that the relex rule file was
> already an egregious disaster when I found it.
>

As I am keenly aware.   Which is why I reacted as I did ..

**Pull request merged**